picard/acoustid/manager.py

```diff
@@ -39,55 +39,83 @@ def __init__(self, fingerprint, duration, orig_recordingid=None, recordingid=Non
         self.puid = puid
         self.orig_recordingid = orig_recordingid
         self.recordingid = recordingid
+        self.attempts = 0
+
+    def __len__(self):
+        # payload approximation
+        # it is based on actual measures, as an example:
+        # with no puid, 2 fingerprints total size of 7501 bytes, post body was 7719 bytes (including all fields)
+        # so that's an overhead of ~3%
+        # we use 10% here, to be safe
+        return int((len(self.fingerprint) + len(self.puid)) * 1.1)
```

Comment:
Should we include `len(self.recording_id) + len(str(self.duration))` here? We have the data available, so helps us to be more accurate.

Reply:
With 10% overhead it isn't needed, but yes if we want to increase precision (not sure if really needed at this point).